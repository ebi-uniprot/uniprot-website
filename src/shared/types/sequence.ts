export type Sequence = {
  value: string;
  length: number;
  molWeight: number;
  /**
   * @deprecated Should use MD5 instead
   */
  crc64: string;
  md5: string;
};
