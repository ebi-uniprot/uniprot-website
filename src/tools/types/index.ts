export enum Tool {
  blast = 'blast',
  // align = 'align',
  // peptideSearch = 'peptideSearch',
  // idMapping = 'idMapping',
}
