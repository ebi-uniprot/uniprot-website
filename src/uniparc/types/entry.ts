export enum TabLocation {
  Entry = 'entry',
  FeatureViewer = 'feature-viewer',
}
