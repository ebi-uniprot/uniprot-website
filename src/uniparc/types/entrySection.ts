export enum EntrySection {
  Sequence = 'sequence',
  XRefs = 'cross-references',
}

export type EntrySectionNameAndId = {
  name: string;
  id: EntrySection;
};

export const getEntrySectionNameAndId = (
  section: EntrySection
  // eslint-disable-next-line consistent-return
): EntrySectionNameAndId => {
  // eslint-disable-next-line default-case
  switch (section) {
    case EntrySection.Sequence:
      return {
        name: 'Sequence',
        id: EntrySection.Sequence,
      };
    case EntrySection.XRefs:
      return {
        name: 'Cross references',
        id: EntrySection.XRefs,
      };
  }
};

export default EntrySection;
