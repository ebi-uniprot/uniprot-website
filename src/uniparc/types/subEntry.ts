export enum TabLocation {
  Entry = 'entry',
  FeatureViewer = 'feature-viewer',
}

enum EntrySection {
  Sequence = 'sequence',
  SimilarProteins = 'similar_proteins',
}

export type EntrySectionNameAndId = {
  name: string;
  id: EntrySection;
};

export default EntrySection;
