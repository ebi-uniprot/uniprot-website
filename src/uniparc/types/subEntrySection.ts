enum SubEntrySection {
  Function = 'function',
  NamesAndTaxonomy = 'names_and_taxonomy',
  SubcellularLocation = 'subcellular_location',
  Interaction = 'interaction',
  Structure = 'structure',
  FamilyAndDomains = 'family_and_domains',
  Sequence = 'sequence',
  SimilarProteins = 'similar_proteins',
  Keywords = 'keywords',
}

export type EntrySectionNameAndId = {
  name: string;
  id: SubEntrySection;
};

export default SubEntrySection;
