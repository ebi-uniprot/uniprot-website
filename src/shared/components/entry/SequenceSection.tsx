import { FC } from 'react';
import { Card, Sequence } from 'franklin-sites';

import { Sequence as SequenceType } from '../../types/sequence';
import { formatLargeNumber, hasContent } from '../../utils/utils';

import EntrySection, {
  getEntrySectionNameAndId,
} from '../../../uniref/types/entrySection';

const SequenceSection: FC<{
  data: SequenceType;
}> = ({ data }) => {
  if (!hasContent(data)) {
    return null;
  }

  const infoData = [
    {
      title: 'Length',
      content: data.length,
    },
    {
      title: 'Mass (Da)',
      content: formatLargeNumber(data.molWeight),
    },
    {
      title: 'Checksum',
      content: data.crc64,
    },
  ];

  return (
    <div id={EntrySection.Sequence}>
      <Card title={getEntrySectionNameAndId(EntrySection.Sequence).name}>
        <Sequence sequence={data.value} infoData={infoData} isCollapsible />
      </Card>
    </div>
  );
};

export default SequenceSection;
