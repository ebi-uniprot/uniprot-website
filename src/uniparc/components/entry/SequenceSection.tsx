import { Card } from 'franklin-sites';

import SimpleSequence from '../../../shared/components/simple-sequence/SimpleSequence';

import { hasContent } from '../../../shared/utils/utils';
import { getEntrySectionNameAndId } from '../../utils/entrySection';

import EntrySection from '../../types/entrySection';
import { Sequence as SequenceType } from '../../../shared/types/sequence';

const SequenceSection = ({ data }: { data?: SequenceType }) => {
  if (!data || !hasContent(data)) {
    return null;
  }

  return (
    <Card
      header={<h2>{getEntrySectionNameAndId(EntrySection.Sequence).name}</h2>}
      id={EntrySection.Sequence}
    >
      <SimpleSequence sequence={data} />
    </Card>
  );
};

export default SequenceSection;
