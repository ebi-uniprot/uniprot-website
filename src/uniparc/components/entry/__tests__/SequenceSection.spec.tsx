import { render } from '@testing-library/react';

import sequenceUIData from '../../../../shared/components/entry/__tests__/__mocks__/sequenceUIData';
import { Sequence as SequenceType } from '../../../../shared/types/sequence';
import SequenceSection from '../SequenceSection';

describe('SequenceSection component', () => {
  it('should render the SequenceSection properly and match snapshot', () => {
    const { asFragment } = render(
      <SequenceSection
        accession="UPI000002DB1C"
        data={sequenceUIData.sequence as SequenceType}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
