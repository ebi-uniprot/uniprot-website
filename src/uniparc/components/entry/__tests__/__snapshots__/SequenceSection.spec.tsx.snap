// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SequenceSection component should render the SequenceSection properly and match snapshot 1`] = `
<DocumentFragment>
  <section
    class="card"
  >
    <div
      class="card__container"
    >
      <div
        class="card__header card__header--with-separator"
      >
        <h2>
          Sequence
        </h2>
      </div>
      <div
        class="card__content"
      >
        <div>
          <section
            class="sequence-container"
          >
            <ul
              class="info-list info-list--columns"
            >
              <li>
                <div
                  class="decorated-list-item decorated-list-item--compact"
                >
                  <div
                    class="decorated-list-item__title tiny"
                  >
                    Length
                  </div>
                  <div
                    class="decorated-list-item__content"
                  >
                    1234
                  </div>
                </div>
              </li>
              <li>
                <div
                  class="decorated-list-item decorated-list-item--compact"
                >
                  <div
                    class="decorated-list-item__title tiny"
                  >
                    Mass (Da)
                  </div>
                  <div
                    class="decorated-list-item__content"
                  >
                    5,678
                  </div>
                </div>
              </li>
              <li>
                <div
                  class="decorated-list-item decorated-list-item--compact"
                >
                  <div
                    class="decorated-list-item__title tiny"
                  >
                    <span
                      data-article-id="checksum"
                    >
                      Checksum
                    </span>
                  </div>
                  <div
                    class="decorated-list-item__content"
                  >
                    ASSDFFFDSAASDDSDFSASDF
                  </div>
                </div>
              </li>
            </ul>
            <div
              class="sequence"
            >
              <span
                class="sequence__chunk"
              >
                ATGHASOADS
              </span>
              <span
                class="sequence__chunk"
              >
                DKDSAJALDS
              </span>
              <span
                class="sequence__chunk"
              >
                AMCMNDSKMD
              </span>
              <span
                class="sequence__chunk"
              >
                SKNCSAKN
              </span>
            </div>
          </section>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
